api/src/main/java/com/datatorrent/api/Context.java

```diff
@@ -161,6 +161,13 @@
      */
     Attribute<Class<?>> TUPLE_CLASS = new Attribute<>(new Class2String<>());
 
+    /**
+     * Attribute of input port.
+     * This is a read-only attribute to query whether the input port is connected to a DelayOperator
+     * This is for iterative processing.
+     */
+    Attribute<Boolean> IS_CONNECTED_TO_DELAY_OPERATOR = new Attribute<>(false);
```

Comment:
I chose to use an attribute because it will be available in child container (namely GenericNode.java).  If there is a way to get to the InputPortMeta in that code, I'd prefer to put it there but it seems that InputPortMeta is only available in stram.